### PR TITLE
Improve the usability of the experiment reports

### PR DESCRIPTION
**Pull request by erfanio**
* The tables are much prettier
* The index.html table has sortable columns
* The index.html table has an index column
* Fix a bug in the signature matching algorithm where it returned early on prefix matches.
* Prevent the licence to show up in the HTML (often multiple times).

Preview: 
image

### CODE REVIEW

**Comment by DonggeLiu** [comment]
Luckily, there does not seem to be a merge conflict : )
I hope you don't mind me taking the liberty to rebase this PR. I was trying to show and fix the merge conflict now, but it seems OK.

**Comment by DonggeLiu** [comment]
A related issues: #143
Happy to discuss more on this tmr

**Comment by DonggeLiu** [comment]
Running a PR-experiment to show the new look : )
JOB: https://console.cloud.google.com/kubernetes/job/us-central1-c/llm-experiment/default/ofg-pr-213-dg
REPORT: https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-04-11-213-dg-comparison/index.html
BUCKET: https://console.cloud.google.com/storage/browser/oss-fuzz-gcb-experiment-run-logs/Result-reports/ofg-pr/2024-04-11-213-dg-comparison
BUCKET GS: `gs://oss-fuzz-gcb-experiment-run-logs/Result-reports/ofg-pr/2024-04-11-213-dg-comparison`

**Comment by DonggeLiu** [comment]
@erfanio Would you mind me merging this, or is it still WIP?
There are two experiments taking very long but they do not affect the new pages looking great.

**Comment by erfanio** [comment]
Feel free to merge. I had forgotten to ask for review.

File: report/web.py [hunk 1]
```diff
@@ -50,32 +50,41 @@ class Benchmark:
...
+
+        # Best match is a full match, but sometimes the result directory only
+        # has the first n characters of a long function name so a full match is
+        # not possible.
+        # To avoid returning early on a prefix match when there is a full match
+        # farther down the list, we only return the prefix match at the end.
+        if function_name.lower() == self.function.lower():
+          return function_signature
+        if function_name.lower().startswith(self.function.lower()):
+          matched_prefix_signature = function_signature
```
Comment (DonggeLiu) [inline report/web.py:85]: Thanks, this is clearer and more robust. 
1. Do you happen to notice any project/function only has a partial match? I don't recall which one it was.
2. Given now we handle this more carefully, shall we log a warning if there are multiple partial matches?

**Review by DonggeLiu — COMMENTED** [review]
Thanks! Two quick questions:

File: report/web.py (hunk 1 above)
Comment (erfanio) [reply report/web.py:85]: https://github.com/google/oss-fuzz-gen/blob/main/benchmark-sets/all/ada-url.yaml

has both `ada_can_parse` and `ada_can_parse_with_base` . I only noticed it when I was looking for benchmarks with the same signature. These two were showing the same signature in the report and I tracked it down to this.

Comment (erfanio) [reply report/web.py:85]: A lot of the FI generated ones like immer are too long so the directories only have a substring.

I think this is the bit where long function names are truncated https://github.com/google/oss-fuzz-gen/blob/main/experiment/benchmark.py#L91

Comment (DonggeLiu) [reply report/web.py:85]: Ah yes you are right.

Comment (erfanio) [reply report/web.py:85]: I think this is further motivation for #105

Comment (DonggeLiu) [reply report/web.py:85]: Yep!

**Review by DonggeLiu — APPROVED** [review]

**Pull request merged**